UnitTests/CommonTestUtils/AsyncTestHelper.cs

```diff
@@ -1,39 +1,48 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
+using System.Diagnostics;
+using System.Threading;
 using System.Threading.Tasks;
 using GitUI;
 
 namespace CommonTestUtils
 {
     public static class AsyncTestHelper
     {
-        public static void RunAndWaitForPendingOperations(Func<Task> asyncMethod)
+        public static TimeSpan UnexpectedTimeout
+        {
+            get
+            {
+                return Debugger.IsAttached ? TimeSpan.FromHours(1) : TimeSpan.FromMinutes(1);
+            }
+        }
+
+        public static void RunAndWaitForPendingOperations(Func<Task> asyncMethod, CancellationToken cancellationToken)
         {
             ThreadHelper.JoinableTaskFactory.Run(asyncMethod);
 
-            WaitForPendingOperations();
+            WaitForPendingOperations(cancellationToken);
         }
 
-        public static T RunAndWaitForPendingOperations<T>(Func<Task<T>> asyncMethod)
+        public static T RunAndWaitForPendingOperations<T>(Func<Task<T>> asyncMethod, CancellationToken cancellationToken)
         {
             var result = ThreadHelper.JoinableTaskFactory.Run(asyncMethod);
 
-            WaitForPendingOperations();
+            WaitForPendingOperations(cancellationToken);
 
             return result;
         }
 
-        public static void WaitForPendingOperations()
+        public static void WaitForPendingOperations(TimeSpan timeout)
         {
-            try
-            {
-                ThreadHelper.JoinableTaskContext?.Factory.Run(() => ThreadHelper.JoinPendingOperationsAsync());
-            }
-            catch (OperationCanceledException)
```

Comment:
💭 I'm not sure why this `catch` handler was originally added, but we can't catch this exception or it will hide potential bugs.